travisci/resource_travisci_environment_variable.go

```
@@ -0,0 +1,92 @@
+		Read:   resourceTravisciEnvironmentVariableRead,
+		Delete: resourceTravisciEnvironmentVariableDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"public": &schema.Schema{
+				Type:     schema.TypeBool,
+				Required: true,
+				ForceNew: true,
+			},
+			"repository_id": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"value": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceTravisciEnvironmentVariableCreate(d *schema.ResourceData, meta interface{}) error {
+	client, err := meta.(*Config).Client()
+
+	envVar := travis.EnvironmentVariable{Name: d.Get("name").(string), Value: d.Get("value").(string), Public: d.Get("public").(bool)}
+	result, _, err := client.EnvironmentVariables.Create(uint(d.Get("repository_id").(int)), &envVar)
+	if err != nil {
+		return err
+	}
+	d.SetId(result.Id)
+
+	return nil
+}
+
+func resourceTravisciEnvironmentVariableRead(d *schema.ResourceData, meta interface{}) error {
+	client, err := meta.(*Config).Client()
+
+	result, _, err := client.EnvironmentVariables.Get(d.Id(), uint(d.Get("repository_id").(int)))
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	if result != nil {
+		d.Set("name", result.Name)
+		d.Set("public", result.Public)
+		d.Set("repository_id", result.RepositoryId)
+		// ignore non-pubic values because the API returns a placeholder value
+		if result.Public {
+			d.Set("value", result.Value)
+		}
+	}
+
+	return nil
+}
+
+func resourceTravisciEnvironmentVariableDelete(d *schema.ResourceData, meta interface{}) error {
+	client, err := meta.(*Config).Client()
+
+	envVar := travis.EnvironmentVariable{Id: d.Id()}
+	_, _, err = client.EnvironmentVariables.Delete(uint(d.Get("repository_id").(int)), &envVar)
+	if err != nil {
+		fmt.Println(err)
```

should this return an error (instead of printing it)?